Sources/StreamChat/StateLayer/Coordinators/ChannelListQueryLinkingCoordinator.swift

```diff
@@ -0,0 +1,100 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// Listens to local events and adds or removes channels from ``ChannelListQueryDTO.channels``.
+@available(iOS 13.0, *)
+struct ChannelListQueryLinkingCoordinator {
```

Comment:
This is the same linking/unlinking logic we have in the controller. Needs some rethinking in next iterations.